### DISCUSSION TITLE
how to invoke a multicall with methods as args

### DISCUSSION THREAD

**Discussion by jasan-s** [wevm/wagmi#854] [Q&A]
in order to reduce requests to my node provider i added [multicall](https://docs.openzeppelin.com/contracts/4.x/utilities#multicall) functionality to my contract, which adds ability to invoke multiple methods in one call and all state changes are handled atomically. However, im unclear how to call multicall method on my contract using wagmi through passing multiple contract methods as args in [useContractRead](https://wagmi.sh/docs/hooks/useContractRead). 

Additionally,  can someone confirm that the way [useContractReads](https://wagmi.sh/docs/hooks/useContractReads) runs is different from [multicall](https://docs.openzeppelin.com/contracts/4.x/utilities#multicall) from open zeppelin. i.e is using `useContractReads` read as a single request or multiple by node provider?  My main motivation is to make a single request to read 10 state vars from smart contract.

**Comment by jasan-s** [comment]
seems [useContractReads](https://wagmi.sh/docs/hooks/useContractReads) batches requests and in my Node provider(Alchemy) the 10 requests are treated as a single request 😄 . So instead of creating a multicall in my contract i decided to use `useContractReads` .  However I would still like to know how i can use a multicall method from contract like mentioned above using wagmi. because i do like the ability to call multiple functions atomically.

**Comment by tmm** [comment]
wagmi uses [multicall3](https://github.com/mds1/multicall)

**Comment by kombos** [comment]
any solutions on this ? how to perform the write operation of multicall function which is available on several contracts ?
